htmd/qm/base.py

```diff
@@ -160,13 +160,11 @@ def _submit(self):
         for directory in self._directories:
             if not self._completed(directory):
                 self.queue.submit(directory)
+        self.queue._dirs = self._directories
 
     def _retrieve(self):
 
-        if isinstance(self.queue, LocalCPUQueue):
-            self.queue.wait(sentinel=False) # TODO: sentinel is still broken
-        else:
-            self.queue.wait(sentinel=True)
+        self.queue.wait(sentinel=True)
```

Comment:
According to the failed tests, `LocalQueue` with `sentinel=True` is still broken. Two weeks  ago I was looking into that, but no clue what is wrong... :(

Reply:
I made it work on my machine, I can't understand what is going on on Travis. Only two tests fail.

Reply:
`LocalQueue` is using threads, so we might have the rate race as it behaves differently. The failing tests are the ones, which actually run QM, not just restart.

Reply:
The failing tests are:
test_h2o2_min and test_h2o2_esp in parameterization/test.py
test_restrained_dihedrals in qm/test.py
I'm really puzzled on why they are failing on Travis but passing on my machine

Reply:
rate race?

Reply:
the first two ones are single QM jobs, how is there a race?

Reply:
There is an additional thread create even for a single job, so it can interfere with the main thread.

Reply:
Let's fix this tomorrow.